shithead_game_server/src/messages.rs

```diff
@@ -35,8 +37,20 @@ pub enum ServerMessage {
 
     InitialCards {
         cards_in_hand: Vec<CardId>,
-        three_up_cards: Vec<CardId>,
     },
+
+    MovePlayerCardFromThreeUpToHand {
+        up_three_card_index: usize,
+    },
+
+    MovePlayerCardFromHandToThreeUp {
+        hand_card_index: usize,
+    },
```

Comment:
You should send the player's id too, so the user would know who changed the card location.